apps/emqx_authn/src/emqx_authn.erl

```diff
@@ -126,3 +133,40 @@ get_enabled_authns() ->
 
 tally_authenticators(#{id := AuthenticatorName}, Acc) ->
     maps:update_with(AuthenticatorName, fun(N) -> N + 1 end, 1, Acc).
+
+%%------------------------------------------------------------------------------
+%% Data backup
+%%------------------------------------------------------------------------------
+
+-define(IMPORT_OPTS, #{rawconf_with_defaults => true, override_to => cluster}).
+
+import_data(RawConf) ->
+    AuthnList = maps:get(<<"authentication">>, RawConf, []),
+    {Changed, Results} = lists:foldr(
+        fun(Authn, {ChangedAcc, ResultsAcc}) ->
+            Id = emqx_authentication:authenticator_id(Authn),
+            Path = [?CONF_NS_ATOM, Id],
+            case update_conf(update_req(Id, Authn)) of
+                {error, {_, _, {not_found, _}}} ->
+                    {ChangedAcc, ResultsAcc#{Path => update_conf(create_req(Authn))}};
+                Res ->
+                    {[Path | ChangedAcc], ResultsAcc#{Path => Res}}
+            end
+        end,
+        {[], #{}},
+        AuthnList
```

Comment:
EMQX schema also supports individual authenticator config as "authentication" value, so it may appear in the raw config: https://github.com/emqx/emqx/blob/b5f24c4f8899315f9d442cb8eac587f88219d680/apps/emqx/src/emqx_schema.erl#L2872
Do we handle this?

Reply:
Added a helper to handle it, thanks!